5] 2001/main.py

```diff
@@ -0,0 +1,128 @@
+from random import randint
+
+
+def calculation(x: int, y: int) -> int:
+    """
+    Vnořená funkce obsahující základní vzorec 
+    pro výpočet hodu kostkou bez modifikátoru.
+    """
+    return x * (randint(1, y+1))
+
+
+def second_cut_check(first_cut_1: str, math_mark_str: int):
+    """
+    Vnořená funkce pro vyhodnocení druhé části zadané 
+    hodnoty, pokud byl použit modifikátor. 
+    Pokud byla hodnota zadáná správně, funce vrátí tuple
+    obsahující druh kostky a modifikátor jako int.
+    Pokud byla hodnota zadána špatně, funkce vrátí None.
+    """
+    second_cut = first_cut_1.split(math_mark_str)
+    if len(second_cut) != 2:
+        return
+
+    try:
+        dice = int(second_cut[0])
+        modifier = int(second_cut[1])
+        return dice, modifier  
+    except ValueError:
+        return
+    
+
+def dice_calculation(entered_value: str):
+    """
+    Funkce pro vyhodnocení hodů kostkou, který uživatel zadá
+    ve formátu xDy+z kde:
+    x = roll (počet hodů)
+    y = dice (kolikastěnná je kostka)
+    z = modifier (hodnota pro mat. operaci, která se má
+                    s výsledkem hodů provést)
+
+    Fukce zkontroluje správnost zadané hodnoty a následně 
+    pomocí funkce calculation() vypočítá a vrátí výsledek
+    ve formátu int.
+    Pokud bude hodnota zadána špatně, vrátí uživateli chybovou
+    hlášku.
+    """
+    error_allert = "WRONG VALUE: a correct value is in format xDy+z."
+    roll = 0
+    dice = 0
+    modifier = 0
+    math_mark = ""
+
+    # first format check
+    if "D" not in entered_value:
```

Comment:
cela ta logika na parsovani vstupu je super komplextni :) 

to jde udelat celkme jednoduse pres regex

Reply:
O Regex slyším prvně, to se teď hned nestihnu doučit, tak jsem opravila bez toho.